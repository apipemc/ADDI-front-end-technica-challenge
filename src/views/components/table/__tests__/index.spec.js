import React from 'react';
import { shallow } from 'enzyme';

import Table from '../index';

describe('Table', () => {
  const MOCK_COLUMNS = [
    {
      titles: ['Full Name'],
      paths: [''],
      renderer: items => (
        <span>{`${items.item.first_name} ${items.item.last_name}`}</span>
      ),
    },
    {
      titles: ['Email'],
      paths: ['email'],
    },
    {
      titles: ['Document Type'],
      paths: ['document_type'],
    },
    {
      titles: ['Document'],
      paths: ['document_id'],
    },
    {
      titles: ['Score'],
      paths: ['score'],
    },
    {
      titles: ['Status'],
      paths: ['status'],
    },
  ];

  const MOCK_DATA = [
    {
      address: 'Cra 46A #40AS -16',
      approved_judicial_past: false,
      approved_personal_information: true,
      document_id: '1095804909',
      document_type: 'CC',
      email: 'andrepipemc@gmail.com',
      first_name: 'Andres',
      last_name: 'Felipe',
      score: 0,
      status: 'Lost',
      _id: '5d180279b4e857d8c303edeb',
    },
  ];

  it('should render a view', () => {
    const wrapper = shallow(
      <Table loading columns={MOCK_COLUMNS} data={MOCK_DATA} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a view loading', () => {
    const wrapper = shallow(<Table loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
