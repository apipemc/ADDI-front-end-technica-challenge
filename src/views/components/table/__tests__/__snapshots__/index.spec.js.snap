// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render a view 1`] = `
<table
  className="table"
>
  <thead
    className="table__thead"
  >
    <tr
      className="table__thead-tr"
    >
      <th
        className="table__th--col"
        key="th-0"
        scope="col"
        style={
          Object {
            "width": "16.666666666666668%",
          }
        }
      >
        Full Name
      </th>
      <th
        className="table__th--col"
        key="th-1"
        scope="col"
        style={
          Object {
            "width": "16.666666666666668%",
          }
        }
      >
        Email
      </th>
      <th
        className="table__th--col"
        key="th-2"
        scope="col"
        style={
          Object {
            "width": "16.666666666666668%",
          }
        }
      >
        Document Type
      </th>
      <th
        className="table__th--col"
        key="th-3"
        scope="col"
        style={
          Object {
            "width": "16.666666666666668%",
          }
        }
      >
        Document
      </th>
      <th
        className="table__th--col"
        key="th-4"
        scope="col"
        style={
          Object {
            "width": "16.666666666666668%",
          }
        }
      >
        Score
      </th>
      <th
        className="table__th--col"
        key="th-5"
        scope="col"
        style={
          Object {
            "width": "16.666666666666668%",
          }
        }
      >
        Status
      </th>
    </tr>
  </thead>
  <tbody
    className="table__tbody"
  >
    <tr
      className="table__tbody-tr"
    >
      <td
        className="table__td"
        colSpan={6}
      >
        <onlyUpdateForKeys(Loader)
          color="#000000"
          css={Object {}}
          loading={true}
          size={35}
          sizeUnit="px"
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table should render a view loading 1`] = `
<table
  className="table"
>
  <thead
    className="table__thead"
  >
    <tr
      className="table__thead-tr"
    />
  </thead>
  <tbody
    className="table__tbody"
  >
    <tr
      className="table__tbody-tr"
    >
      <td
        className="table__td"
        colSpan={0}
      >
        <onlyUpdateForKeys(Loader)
          color="#000000"
          css={Object {}}
          loading={true}
          size={35}
          sizeUnit="px"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
